### PR TITLE
Add glitch0r plugin

### PR DESCRIPTION
**Pull request by namikiri**
I love glitches. Glitches love me. Why we shouldn't make a nice effect for such an interesting framew0rk? 

Thank you for inspiration, frei0r maintainers. The video on frei0r.dyne.org made me happy and inspired me to write this little plugin.

Hope this will be helpful :3

### CODE REVIEW

**Comment by jaromil** [comment]
hi, thanks for the contribution! can you rebase all commits into a single one?

**Comment by namikiri** [comment]
@jaromil Yes, of course. Just did it. 

I am new to Github community and I am planning to create another plugin, should I rebase further commits again before making a new pull req?

**Comment by ddennedy** [comment]
The GitHub web UI makes it easy to squash commits or rebase and merge. However, your rebase does make it easier to review and add comments to lines.

File: src/filter/glitch0r/glitch0r.c [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+
+#include <stdlib.h>
+#include <assert.h>
+#include <string.h>
+#include <time.h>
+
+#include "frei0r.h"
+
+/* cheap & fast randomizer (by Fukuchi Kentarou) */
+uint32_t randval;
```
Comment (ddennedy) [inline src/filter/glitch0r/glitch0r.c]: Please make all private symbols static. Names such as fastrand and fastsrand have a strong chance of collision.

File: src/filter/glitch0r/glitch0r.c [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+    glitch0rInfo->num_params =  4; 
+    glitch0rInfo->explanation = "Adds glitches and block shifting";
+}
+
+void f0r_get_param_info(f0r_param_info_t* info, int param_index)
+{
+    switch(param_index) {
+        case 0:
+        {
+            info->name = "GlitchFrequency";
```
Comment (ddennedy) [inline src/filter/glitch0r/glitch0r.c]: This is not a big deal, but some frei0r plugin hosts expose these names as labels in the UI. Thus, spaces are preferred over PascalCase.

**Comment by ddennedy** [comment]
Thank you for your contribution, @namikiri !

**Comment by namikiri** [comment]
Thank you @ddennedy for your review!
I've fixed everything that you pointed in my code.

First time I thought that glitching means total data distortion, but then I tried saving alpha channel and it really makes sense. Thank you for pointing me on it.

Should I squash this commit too?

**Pull request merged**